app/controllers/events_controller.rb

```diff
@@ -0,0 +1,74 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :update, :destroy]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @events = Event.all
+  end
+
+  # GET /events/1
+  # GET /events/1.json
+  def show
+  end
+
+  # GET /events/new
+  def new
+    @event = Event.new
+  end
+
+  # GET /events/1/edit
+  def edit
+  end
+
+  # POST /events
+  # POST /events.json
+  def create
+    @event = Event.new(event_params)
+
+    respond_to do |format|
+      if @event.save
+        format.html { redirect_to @event, notice: 'Event was successfully created.' }
+        format.json { render :show, status: :created, location: @event }
```

Comment:
Especially here, I'd really like to know what this json section is doing.

Reply:
I don't have a super good overview... I think that if the browser requests html it returns lines 31 and if it requests JSON it returns 32. We should dig into it further together.